pkg/cluster/config_test.go

```diff
@@ -188,3 +188,63 @@ func TestGetVXLANPort(t *testing.T) {
 		})
 	}
 }
+
+func TestGetClusterDNS(t *testing.T) {
+	type args struct {
+		iprange string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "empty iprange",
+			args:    args{iprange: ""},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "invalid iprange",
+			args:    args{iprange: "invalid"},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "no ip in range",
+			args:    args{iprange: "172.30.0.0/32"},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name:    "valid iprange",
+			args:    args{iprange: "172.30.0.0/16"},
+			want:    "172.30.0.10",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetClusterDNS(tt.args.iprange)
+			// check err
+			if err != nil {
+				// expect err
+				if !tt.wantErr {
+					t.Errorf("GetClusterDNS() error = %v, wantErr %v", err, tt.wantErr)
+					return
+				}
+				// check err msg
+				errMsg := err.Error()
+				t.Logf("error = %s", errMsg)
+				if errMsg == "" {
+					t.Errorf("failed to detect error message")
+					return
+				}
```

Comment:
Why is this needed?

Reply:
Agree with @aravindhp  comment. if check is not required. if you still want to explicitly mentioned that "failed to detect error message shouldn't be like this 

                                errMsg := err.Error()
				if errMsg == "" {
					t.Errorf("failed to detect error message")
				} else {
				        t.Logf("error = %s", errMsg)
				}

Reply:
> Why is this needed?

To ensure the error message is valid (non-empty).

>Agree with @aravindhp comment. if check is not required. if you still want to explicitly mentioned that "failed to detect error message shouldn't be like this

Open to discuss whether it is required or not.

Reply:
>To ensure the error message is valid (non-empty).

What is the point of checking that it is just non-empty. You need to add `wantErrMsg` and ensure it is the expected error message in case of error.

Reply:
Correct, adding the content of the expected error message (`wantErrMsg`) will fully cover this test around the error message. Ideally, the content of the error message would be loaded from a resource or constant file (require recompilation) that serves as the central source, but it is not the case. So, will drop the check to avoid any confusion .